src/Ogmios.purs

```diff
@@ -238,3 +261,70 @@ messageFoldF
 messageFoldF msg acc' func = do
   acc <- acc'
   if isRight acc then acc' else func msg
+
+--------------------------------------------------------------------------------
+-- Ogmios functions and types to internal types
+--------------------------------------------------------------------------------
+-- JsonWsp.Address is a bech32 string, so wrap to Transaction.Types.Bech32
+-- | Converts an JsonWsp.Address to (internal) Address
+ogmiosAddressToAddress :: JsonWsp.Address -> Effect (Maybe Address)
+ogmiosAddressToAddress ogAddr =
+  newAddressFromBech32 (wrap ogAddr) <#> Deserialization.convertAddress
+
+-- | Converts an (internal) Address to JsonWsp.Address
+addressToOgmiosAddress :: Address -> Effect (Maybe JsonWsp.Address)
+addressToOgmiosAddress addr = do
+  addr' <- Serialization.convertAddress addr
+  pure $ newBaseAddressFromAddress addr' >>= addressPubKeyHash <#> unwrap
+
+-- If required, we can change to Either with more granular error handling.
+-- | Gets utxos at an (internal) Address in terms of (internal) Transaction.Types
+utxosAt' :: Address -> QueryM (Maybe UtxoM)
+utxosAt' addr = do
+  addr' :: Maybe JsonWsp.Address <-
```

Comment:
This is a stylistic thing and my personal preference, and I am of course open to debating it: I understand type signatures on `where` clauses and why our Haskell style guide mandates them (`let` bindings less so, but it can sometimes make things clearer and I wouldn't recommend removing them if they're already there). I would prefer, however, to only use type signatures on local bindings if the types cannot be inferred (this also applies to lines 284, 293, 308, in this module), mostly for the sake of readability

Reply:
So I tend to do it to be explicit - it assists with readability and shouldn't make it worse, although it makes things less terse. I think 284, 308 (and 322 for that matter) are fair, as they're easy to infer. Although 293 is a bit more complex and its type is somewhat useful IMO. I would even say the `let` type expression below is helpful too. Potentially one for the style guide, but I would argue to leave it to the reader's judgement, depending on the binding's complexity?

I've removed 284, 308 and 322 but left the others for now.

Reply:
Fair enough, this is a matter of personal preference. I would add though that the effect on readability is subjective as well, as I find it makes things less readable